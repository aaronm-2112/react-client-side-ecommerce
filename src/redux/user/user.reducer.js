import { UserActionTypes } from "./user.types";

// when we first call this reducer Redux doesn;t have state so give ti a default
const INITIAL_STATE = {
  currentUser: null,
};

// the default value for state is to not be set. We use the INITIAL_STATE for when this is the case.
const userReducer = (currentState = INITIAL_STATE, action) => {
  switch (action.type) {
    case UserActionTypes.SET_CURRENT_USER:
      return {
        ...currentState,
        currentUser: action.payload,
      };
    default:
      return currentState;
  }
};

export default userReducer;
