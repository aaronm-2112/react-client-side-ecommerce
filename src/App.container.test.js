import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import AppContainer from "./App.container";
import { UserActionTypes } from "./redux/user/user.types";

const mockStore = configureMockStore();

describe("MapStateToProps and MapDispatchToProps tests for the App component", () => {
  let store;
  let wrapper;

  beforeEach(() => {
    const initialState = {
      // our user reducer's state
      user: { currentUser: 1 },
    };

    store = mockStore(initialState);
    wrapper = shallow(<AppContainer store={store} />);
  });

  test("App snapshot matches", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test that mapstatetoprops happens correctly
  it("Should pass a value for the user object into the props", () => {
    expect(wrapper.dive().props().currentUser).toBe(1);
  });

  it("Should pass a function for the setCurrentUser action into props", () => {
    let diveWrapper = wrapper.dive();
    expect(diveWrapper.props().setCurrentUser).not.toBe(undefined);
    expect(diveWrapper.props().setCurrentUser()).toEqual({
      payload: undefined,
      type: UserActionTypes.SET_CURRENT_USER,
    });
  });
});
