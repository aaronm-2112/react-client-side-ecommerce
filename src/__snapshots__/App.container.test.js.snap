// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapStateToProps and MapDispatchToProps tests for the App component App snapshot matches 1`] = `
<ContextProvider
  value={null}
>
  <App
    currentUser={1}
    setCurrentUser={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
